executor/cte_test.go

```diff
@@ -440,3 +441,28 @@ func TestCTEExecError(t *testing.T) {
 		require.True(t, terror.ErrorEqual(err, types.ErrOverflow))
 	}
 }
+
+// https://github.com/pingcap/tidb/issues/33965.
+func TestCTEsInView(t *testing.T) {
+	store, clean := testkit.CreateMockStore(t)
+	defer clean()
+	tk := testkit.NewTestKit(t, store)
+	tk.MustExec("use test;")
+	require.True(t, tk.Session().Auth(&auth.UserIdentity{Username: "root", Hostname: "%"}, nil, nil))
```

Comment:
Why do you want to check this?

Reply:
It gives the current session a privilege so that `SELECT` views can pass the check.